libc/test/src/math/smoke/CMakeLists.txt

```diff
@@ -839,6 +839,198 @@ add_fp_unittest(
     libc.src.math.frexpf128
 )
 
+add_fp_unittest(
+  fromfp_test
+  SUITE
+    libc-math-smoke-tests
+  SRCS
+    fromfp_test.cpp
+  HDRS
+    FromfpTest.h
+  DEPENDS
+    libc.src.math.fromfp
```

Comment:
In my previous PR for `nextup` and `nextdown` I had added more dependencies: https://github.com/llvm/llvm-project/pull/85431/files#diff-2269fc16d8717f30b774875846f93f54879dae7515bdfd3dd5d5645fd7413aa7R1626-R1628.

Now that I think about it again, I assume that depending on `libc.src.math.<func>` is enough to also depend on the header and dependencies declared in the `add_entrypoint_object` call for the function in libc/src/math/generic/CMakeLists.txt.

Reply:
Yes, but _include-what-you-use_ is always better, instead of completely relying on transitive dependence.  So if your test directly calls something in those targets, it's always better to directly include them in our DEPENDS list.

Reply:
Right, but the `nextup` and `nextdown` tests only directly use `libc.src.math.{nextup,nextdown}` and test/UnitTest/{FPMatcher.h,Test.h}, and `add_fp_unittest` already takes care of the latter.